backend/sessionstore.go

```diff
@@ -0,0 +1,157 @@
+package backend
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gorilla/sessions"
+)
+
+// SessionStore defines an interface which manages the handling of Session management.
+type SessionStore interface {
+	// Get returns a Session based on the given id.
+	Get(string, ...SessionStoreOption) (Session, error)
```

Comment:
Because the option is providing the request/response interface (through either fiber.Ctx or net/http w/r), these could change with each call to Get or Save, and the values set for the New store will likely no longer be current.

What may be easier is to pass a HTTPHandler, which will have the required methods anyway? Similar to calling sessions.Get(r, ID)

Reply:
Sorry, I explained that poorly, it wouldn't be changing between net/http and fiber.

Consider the following 2 handlers (in net/http)
```go
// handles requests to /first
func firstHandler(w, r) {
    h, _ := newHTTPhandler(backend.NetHTTP, WithHTTPWriterAndRequest(w,r))
    // Here h has the request to /first, and will write responses via w to that request client.
    sessionStore.Get(sessionID, withNetHTTPHandler(w,r))
    // This will get the sessions from the cookies for the request to /first
}

// handles requests to /second
func secondHandler(w, r) {
    h, _ := newHTTPhandler(backend.NetHTTP)
    // Without passing the request and response writer, this either has no context, 
    // or depending on how the handler may get reused with a global service, it may
    // have the request and response writer from the request to /first.
    sessionStore.Get(sessionID)
    // Assuming that the w and r from the first request are still attached, this will not 
    // get the sessions for the current request to /second
}
```

Hopefully this is a little more clear. Basically the fiber.Ctx or responseWriter and Request will change on every new call to a handler, and need to be passed in to calls to the session store to make sure that the correct sessions are actually being gotten. 

This may not be the best way, but consider that the gorilla/sessions implementation does the same thing, but without using functional options.
```golang
func (s *CookieStore) Get(r *http.Request, name string) (*Session, error)
```
We are just doing the same thing, but with functional options to make the interface generalisable over different http handler frameworks.